diff --git a/ld/scripttempl/elf.sc b/ld/scripttempl/elf.sc
index be8d19fc..e7045e09 100644
--- a/ld/scripttempl/elf.sc
+++ b/ld/scripttempl/elf.sc
@@ -790,6 +790,8 @@ EOF
 emit_data()
 {
 cat <<EOF
+  /* Hacky hackity hack. Need this because otherwise data segment is not aligned enough. */
+  . = ALIGN(0x10000);
   /* Exception handling.  */
   .eh_frame     ${RELOCATING-0} : ONLY_IF_RW { KEEP (*(.eh_frame))${RELOCATING+ *(.eh_frame.*)} }
   .sframe       ${RELOCATING-0} : ONLY_IF_RW { *(.sframe)${RELOCATING+ *(.sframe.*)} }
